processor/src/main/java/org/mapstruct/ap/internal/model/NestedTargetPropertyMappingHolder.java

```diff
@@ -180,9 +180,15 @@ public NestedTargetPropertyMappingHolder build() {
                         .groupedBySourceReferences
                         .entrySet() ) {
                         PropertyEntry sourceEntry = entryBySP.getKey();
+                        boolean forceUpdateMethodOrNonNestedReferencesPresent =
+                            forceUpdateMethod || !groupedSourceReferences.nonNested.isEmpty();
+                        // if an update method is forced or there are non nested source references then
+                        // we need to create mapping options for forged methods (i.e. Ignore unmapped target properties
+                        // because it the nonNested properties might contain the target properties)
                         MappingOptions sourceMappingOptions = MappingOptions.forMappingsOnly(
                             groupByTargetName( entryBySP.getValue() ),
-                            forceUpdateMethod
+                            forceUpdateMethod,
```

Comment:
the parameter name is `restrictToDefinedMappings` however we put in`forceUpdateMethod` how do they relate?.  Same goes for the next arg.

Reply:
It was like that before. Maybe we need to rename it to `restrictToDefinedMappings` instead.

Reply:
ok..